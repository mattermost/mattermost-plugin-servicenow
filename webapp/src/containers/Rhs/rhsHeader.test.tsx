import React from 'react';
import {shallow} from 'enzyme';
import * as redux from 'react-redux';

import RHSHeader from './rhsHeader';

describe('RHSHeader', () => {
    const baseProps = {
        showAllSubscriptions: false,
        setShowAllSubscriptions: jest.fn(),
        filter: {createdBy: 'mockUser'},
        setFilter: jest.fn(),
        setResetFilter: jest.fn(),
    };

    // Mock useDispatch hook
    const spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');

    // Mock dispatch function returned from useDispatch
    const mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should match snapshot when filter icon is present', () => {
        const wrapper = shallow(
            <RHSHeader
                {...baseProps}
                showFilterIcon={true}
            />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.text().includes('Subscriptions')).toBeTruthy();
        expect(wrapper.find('ToggleSwitch').exists()).toBeFalsy();
        expect(wrapper.find('MenuButtons').exists()).toBeFalsy();

        wrapper.find('IconButton').first().simulate('click');
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('ToggleSwitch').exists()).toBeTruthy();
        expect(wrapper.find('Dropdown')).toHaveLength(1);
        expect(wrapper.find('Button')).toHaveLength(2);

        wrapper.find('IconButton').last().simulate('click');
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').last().hasClass('rhs-filter-popover rhs-menu-popover'));
        expect(wrapper.find('MenuButtons').exists()).toBeTruthy();

        wrapper.find('Button').first().simulate('click');
        expect(baseProps.setShowAllSubscriptions).toBeCalled();
        expect(baseProps.setFilter).toBeCalled();
        expect(baseProps.setResetFilter).toBeCalled();
    });

    it('should match snapshot when filter icon is not present', () => {
        const wrapper = shallow(
            <RHSHeader
                {...baseProps}
                showFilterIcon={false}
            />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.text().includes('Subscriptions')).toBeTruthy();
        expect(wrapper.find('ToggleSwitch').exists()).toBeFalsy();
        expect(wrapper.find('MenuButtons').exists()).toBeFalsy();

        wrapper.find('IconButton').first().simulate('click');
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').last().hasClass('rhs-filter-popover rhs-menu-popover'));
        expect(wrapper.find('MenuButtons').exists()).toBeTruthy();
        expect(wrapper.find('ToggleSwitch').exists()).toBeFalsy();
        expect(wrapper.find('Dropdown')).toHaveLength(0);
        expect(wrapper.find('Button')).toHaveLength(0);
    });
});
