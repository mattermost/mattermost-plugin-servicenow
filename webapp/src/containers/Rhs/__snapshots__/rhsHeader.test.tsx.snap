// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RHSHeader should match snapshot when filter icon is not present 1`] = `
<Fragment>
  <div
    className="position-relative rhs-header-divider"
  >
    <div
      className="d-flex align-item-center"
    >
      <p
        className="rhs-title"
      >
        Subscriptions
      </p>
      <IconButton
        extraClass="margin-left-auto"
        onClick={[Function]}
        tooltipText="Actions"
      >
        <SVGWrapper
          className="padding-left-6"
          height={28}
          viewBox="0 -1 11 10"
          width={16}
        >
          <path
            d="M2 12C2.53043 12 3.03914 12.2107 3.41421 12.5858C3.78929 12.9609 4 13.4696 4 14C4 14.5304 3.78929 15.0391 3.41421 15.4142C3.03914 15.7893 2.53043 16 2 16C1.46957 16 0.96086 15.7893 0.585787 15.4142C0.210714 15.0391 0 14.5304 0 14C0 13.4696 0.210714 12.9609 0.585787 12.5858C0.96086 12.2107 1.46957 12 2 12ZM2 6C2.53043 6 3.03914 6.21071 3.41421 6.58579C3.78929 6.96086 4 7.46957 4 8C4 8.53043 3.78929 9.03914 3.41421 9.41421C3.03914 9.78929 2.53043 10 2 10C1.46957 10 0.96086 9.78929 0.585787 9.41421C0.210714 9.03914 0 8.53043 0 8C0 7.46957 0.210714 6.96086 0.585787 6.58579C0.96086 6.21071 1.46957 6 2 6ZM2 0C2.53043 0 3.03914 0.210714 3.41421 0.585786C3.78929 0.960859 4 1.46957 4 2C4 2.53043 3.78929 3.03914 3.41421 3.41421C3.03914 3.78929 2.53043 4 2 4C1.46957 4 0.96086 3.78929 0.585787 3.41421C0.210714 3.03914 0 2.53043 0 2C0 1.46957 0.210714 0.960859 0.585787 0.585786C0.96086 0.210714 1.46957 0 2 0Z"
            fill="#90929A"
          />
        </SVGWrapper>
      </IconButton>
    </div>
  </div>
</Fragment>
`;

exports[`RHSHeader should match snapshot when filter icon is not present 2`] = `
<Fragment>
  <div
    className="position-relative rhs-header-divider"
  >
    <div
      className="d-flex align-item-center"
    >
      <p
        className="rhs-title"
      >
        Subscriptions
      </p>
      <IconButton
        extraClass="margin-left-auto"
        onClick={[Function]}
        tooltipText="Actions"
      >
        <SVGWrapper
          className="padding-left-6"
          height={28}
          viewBox="0 -1 11 10"
          width={16}
        >
          <path
            d="M2 12C2.53043 12 3.03914 12.2107 3.41421 12.5858C3.78929 12.9609 4 13.4696 4 14C4 14.5304 3.78929 15.0391 3.41421 15.4142C3.03914 15.7893 2.53043 16 2 16C1.46957 16 0.96086 15.7893 0.585787 15.4142C0.210714 15.0391 0 14.5304 0 14C0 13.4696 0.210714 12.9609 0.585787 12.5858C0.96086 12.2107 1.46957 12 2 12ZM2 6C2.53043 6 3.03914 6.21071 3.41421 6.58579C3.78929 6.96086 4 7.46957 4 8C4 8.53043 3.78929 9.03914 3.41421 9.41421C3.03914 9.78929 2.53043 10 2 10C1.46957 10 0.96086 9.78929 0.585787 9.41421C0.210714 9.03914 0 8.53043 0 8C0 7.46957 0.210714 6.96086 0.585787 6.58579C0.96086 6.21071 1.46957 6 2 6ZM2 0C2.53043 0 3.03914 0.210714 3.41421 0.585786C3.78929 0.960859 4 1.46957 4 2C4 2.53043 3.78929 3.03914 3.41421 3.41421C3.03914 3.78929 2.53043 4 2 4C1.46957 4 0.96086 3.78929 0.585787 3.41421C0.210714 3.03914 0 2.53043 0 2C0 1.46957 0.210714 0.960859 0.585787 0.585786C0.96086 0.210714 1.46957 0 2 0Z"
            fill="#90929A"
          />
        </SVGWrapper>
      </IconButton>
    </div>
  </div>
  <div
    className="rhs-filter-popover rhs-menu-popover"
  >
    <MenuButtons
      buttons={
        Array [
          Object {
            "icon": <SVGWrapper
              height={16}
              viewBox="0 0 17 12"
              width={16}
            >
              <path
                d="M14.4 0H1.6C0.72 0 0 0.72 0 1.6V12.8C0 13.68 0.72 14.4 1.6 14.4H14.4C15.28 14.4 16 13.68 16 12.8V1.6C16 0.72 15.28 0 14.4 0ZM14.4 12.8H1.6V3.2H14.4V12.8ZM6.4 7.2V5.6H12.8V7.2H6.4ZM3.2 5.6H4.8V7.2H3.2V5.6ZM6.4 10.4V8.8H12.8V10.4H6.4ZM3.2 8.8H4.8V10.4H3.2V8.8Z"
                fill="#3F4350"
              />
            </SVGWrapper>,
            "onClick": [Function],
            "text": "Begin catalog request",
          },
          Object {
            "icon": <SVGWrapper
              height={16}
              viewBox="0 0 17 17"
              width={16}
            >
              <path
                d="M9.00002 0.666626C4.41669 0.666626 0.666687 4.41663 0.666687 8.99996C0.666687 13.5833 4.41669 17.3333 9.00002 17.3333C13.5834 17.3333 17.3334 13.5833 17.3334 8.99996C17.3334 4.41663 13.5834 0.666626 9.00002 0.666626ZM9.00002 15.6666C5.33335 15.6666 2.33335 12.6666 2.33335 8.99996C2.33335 5.33329 5.33335 2.33329 9.00002 2.33329C12.6667 2.33329 15.6667 5.33329 15.6667 8.99996C15.6667 12.6666 12.6667 15.6666 9.00002 15.6666ZM9.41752 9.83329H8.58419L8.16752 4.83329H9.83419L9.41752 9.83329ZM9.83335 12.3333C9.83335 12.7933 9.46002 13.1666 9.00002 13.1666C8.54002 13.1666 8.16669 12.7933 8.16669 12.3333C8.16669 11.8733 8.54002 11.5 9.00002 11.5C9.46002 11.5 9.83335 11.8733 9.83335 12.3333Z"
                fill="#3F4350"
              />
            </SVGWrapper>,
            "onClick": [Function],
            "text": "Create an incident",
          },
          Object {
            "icon": <SVGWrapper
              height={14}
              viewBox="0 2 16 10"
              width={14}
            >
              <path
                d="M11.6667 10.9511C11.0756 10.9511 10.5467 11.1844 10.1422 11.55L4.59667 8.32222C4.63556 8.14333 4.66667 7.96444 4.66667 7.77778C4.66667 7.59111 4.63556 7.41222 4.59667 7.23333L10.08 4.03667C10.5 4.42556 11.0522 4.66667 11.6667 4.66667C12.9578 4.66667 14 3.62444 14 2.33333C14 1.04222 12.9578 0 11.6667 0C10.3756 0 9.33333 1.04222 9.33333 2.33333C9.33333 2.52 9.36444 2.69889 9.40333 2.87778L3.92 6.07444C3.5 5.68556 2.94778 5.44444 2.33333 5.44444C1.04222 5.44444 0 6.48667 0 7.77778C0 9.06889 1.04222 10.1111 2.33333 10.1111C2.94778 10.1111 3.5 9.87 3.92 9.48111L9.45778 12.7089C9.41889 12.8722 9.39556 13.0433 9.39556 13.2222C9.39556 14.4744 10.4144 15.4856 11.6667 15.4856C12.9189 15.4856 13.9378 14.4744 13.9378 13.2222C13.9378 11.97 12.9189 10.9511 11.6667 10.9511ZM11.6667 1.55556C12.0944 1.55556 12.4444 1.90556 12.4444 2.33333C12.4444 2.76111 12.0944 3.11111 11.6667 3.11111C11.2389 3.11111 10.8889 2.76111 10.8889 2.33333C10.8889 1.90556 11.2389 1.55556 11.6667 1.55556ZM2.33333 8.55556C1.90556 8.55556 1.55556 8.20555 1.55556 7.77778C1.55556 7.35 1.90556 7 2.33333 7C2.76111 7 3.11111 7.35 3.11111 7.77778C3.11111 8.20555 2.76111 8.55556 2.33333 8.55556ZM11.6667 14C11.2389 14 10.8889 13.65 10.8889 13.2222C10.8889 12.7944 11.2389 12.4444 11.6667 12.4444C12.0944 12.4444 12.4444 12.7944 12.4444 13.2222C12.4444 13.65 12.0944 14 11.6667 14Z"
                fill="#3F4350"
              />
            </SVGWrapper>,
            "onClick": [Function],
            "text": "Share a record",
          },
        ]
      }
    />
  </div>
</Fragment>
`;

exports[`RHSHeader should match snapshot when filter icon is present 1`] = `
<Fragment>
  <div
    className="position-relative rhs-header-divider"
  >
    <div
      className="d-flex align-item-center"
    >
      <p
        className="rhs-title"
      >
        Subscriptions
      </p>
      <IconButton
        extraClass="margin-left-auto flex-basis-initial margin-right-8 filter-button"
        onClick={[Function]}
        tooltipText="Filter"
      >
        <SVGWrapper
          height={12}
          viewBox="0 0 18 12"
          width={18}
        >
          <path
            d="M3 7H15V5H3V7ZM0 0V2H18V0H0ZM7 12H11V10H7V12Z"
            fill="#8A8C94"
          />
        </SVGWrapper>
      </IconButton>
      <IconButton
        extraClass=""
        onClick={[Function]}
        tooltipText="Actions"
      >
        <SVGWrapper
          className="padding-left-6"
          height={28}
          viewBox="0 -1 11 10"
          width={16}
        >
          <path
            d="M2 12C2.53043 12 3.03914 12.2107 3.41421 12.5858C3.78929 12.9609 4 13.4696 4 14C4 14.5304 3.78929 15.0391 3.41421 15.4142C3.03914 15.7893 2.53043 16 2 16C1.46957 16 0.96086 15.7893 0.585787 15.4142C0.210714 15.0391 0 14.5304 0 14C0 13.4696 0.210714 12.9609 0.585787 12.5858C0.96086 12.2107 1.46957 12 2 12ZM2 6C2.53043 6 3.03914 6.21071 3.41421 6.58579C3.78929 6.96086 4 7.46957 4 8C4 8.53043 3.78929 9.03914 3.41421 9.41421C3.03914 9.78929 2.53043 10 2 10C1.46957 10 0.96086 9.78929 0.585787 9.41421C0.210714 9.03914 0 8.53043 0 8C0 7.46957 0.210714 6.96086 0.585787 6.58579C0.96086 6.21071 1.46957 6 2 6ZM2 0C2.53043 0 3.03914 0.210714 3.41421 0.585786C3.78929 0.960859 4 1.46957 4 2C4 2.53043 3.78929 3.03914 3.41421 3.41421C3.03914 3.78929 2.53043 4 2 4C1.46957 4 0.96086 3.78929 0.585787 3.41421C0.210714 3.03914 0 2.53043 0 2C0 1.46957 0.210714 0.960859 0.585787 0.585786C0.96086 0.210714 1.46957 0 2 0Z"
            fill="#90929A"
          />
        </SVGWrapper>
      </IconButton>
    </div>
  </div>
</Fragment>
`;

exports[`RHSHeader should match snapshot when filter icon is present 2`] = `
<Fragment>
  <div
    className="position-relative rhs-header-divider"
  >
    <div
      className="d-flex align-item-center"
    >
      <p
        className="rhs-title"
      >
        Subscriptions
      </p>
      <IconButton
        extraClass="margin-left-auto flex-basis-initial margin-right-8 filter-button"
        onClick={[Function]}
        tooltipText="Filter"
      >
        <SVGWrapper
          height={12}
          viewBox="0 0 18 12"
          width={18}
        >
          <path
            d="M3 7H15V5H3V7ZM0 0V2H18V0H0ZM7 12H11V10H7V12Z"
            fill="#8A8C94"
          />
        </SVGWrapper>
      </IconButton>
      <IconButton
        extraClass=""
        onClick={[Function]}
        tooltipText="Actions"
      >
        <SVGWrapper
          className="padding-left-6"
          height={28}
          viewBox="0 -1 11 10"
          width={16}
        >
          <path
            d="M2 12C2.53043 12 3.03914 12.2107 3.41421 12.5858C3.78929 12.9609 4 13.4696 4 14C4 14.5304 3.78929 15.0391 3.41421 15.4142C3.03914 15.7893 2.53043 16 2 16C1.46957 16 0.96086 15.7893 0.585787 15.4142C0.210714 15.0391 0 14.5304 0 14C0 13.4696 0.210714 12.9609 0.585787 12.5858C0.96086 12.2107 1.46957 12 2 12ZM2 6C2.53043 6 3.03914 6.21071 3.41421 6.58579C3.78929 6.96086 4 7.46957 4 8C4 8.53043 3.78929 9.03914 3.41421 9.41421C3.03914 9.78929 2.53043 10 2 10C1.46957 10 0.96086 9.78929 0.585787 9.41421C0.210714 9.03914 0 8.53043 0 8C0 7.46957 0.210714 6.96086 0.585787 6.58579C0.96086 6.21071 1.46957 6 2 6ZM2 0C2.53043 0 3.03914 0.210714 3.41421 0.585786C3.78929 0.960859 4 1.46957 4 2C4 2.53043 3.78929 3.03914 3.41421 3.41421C3.03914 3.78929 2.53043 4 2 4C1.46957 4 0.96086 3.78929 0.585787 3.41421C0.210714 3.03914 0 2.53043 0 2C0 1.46957 0.210714 0.960859 0.585787 0.585786C0.96086 0.210714 1.46957 0 2 0Z"
            fill="#90929A"
          />
        </SVGWrapper>
      </IconButton>
    </div>
  </div>
  <div
    className="rhs-filter-popover"
  >
    <div
      className="d-flex align-item-center margin-bottom-15 toggle-class"
    >
      <ToggleSwitch
        active={false}
        label="Show all subscriptions"
        labelPositioning="right"
        onChange={[Function]}
      />
    </div>
    <div
      className="margin-bottom-15"
    >
      <Dropdown
        disabled={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Me",
              "value": "me",
            },
            Object {
              "label": "Anyone",
              "value": "anyone",
            },
          ]
        }
        placeholder="Created By"
        value="mockUser"
      />
    </div>
    <div
      className="text-align-right"
    >
      <Button
        extraClass="margin-right-8"
        isDisabled={false}
        isSecondaryButton={true}
        onClick={[Function]}
        text="Reset"
      />
      <Button
        onClick={[Function]}
        text="Close"
      />
    </div>
  </div>
</Fragment>
`;

exports[`RHSHeader should match snapshot when filter icon is present 3`] = `
<Fragment>
  <div
    className="position-relative rhs-header-divider"
  >
    <div
      className="d-flex align-item-center"
    >
      <p
        className="rhs-title"
      >
        Subscriptions
      </p>
      <IconButton
        extraClass="margin-left-auto flex-basis-initial margin-right-8 filter-button"
        onClick={[Function]}
        tooltipText="Filter"
      >
        <SVGWrapper
          height={12}
          viewBox="0 0 18 12"
          width={18}
        >
          <path
            d="M3 7H15V5H3V7ZM0 0V2H18V0H0ZM7 12H11V10H7V12Z"
            fill="#8A8C94"
          />
        </SVGWrapper>
      </IconButton>
      <IconButton
        extraClass=""
        onClick={[Function]}
        tooltipText="Actions"
      >
        <SVGWrapper
          className="padding-left-6"
          height={28}
          viewBox="0 -1 11 10"
          width={16}
        >
          <path
            d="M2 12C2.53043 12 3.03914 12.2107 3.41421 12.5858C3.78929 12.9609 4 13.4696 4 14C4 14.5304 3.78929 15.0391 3.41421 15.4142C3.03914 15.7893 2.53043 16 2 16C1.46957 16 0.96086 15.7893 0.585787 15.4142C0.210714 15.0391 0 14.5304 0 14C0 13.4696 0.210714 12.9609 0.585787 12.5858C0.96086 12.2107 1.46957 12 2 12ZM2 6C2.53043 6 3.03914 6.21071 3.41421 6.58579C3.78929 6.96086 4 7.46957 4 8C4 8.53043 3.78929 9.03914 3.41421 9.41421C3.03914 9.78929 2.53043 10 2 10C1.46957 10 0.96086 9.78929 0.585787 9.41421C0.210714 9.03914 0 8.53043 0 8C0 7.46957 0.210714 6.96086 0.585787 6.58579C0.96086 6.21071 1.46957 6 2 6ZM2 0C2.53043 0 3.03914 0.210714 3.41421 0.585786C3.78929 0.960859 4 1.46957 4 2C4 2.53043 3.78929 3.03914 3.41421 3.41421C3.03914 3.78929 2.53043 4 2 4C1.46957 4 0.96086 3.78929 0.585787 3.41421C0.210714 3.03914 0 2.53043 0 2C0 1.46957 0.210714 0.960859 0.585787 0.585786C0.96086 0.210714 1.46957 0 2 0Z"
            fill="#90929A"
          />
        </SVGWrapper>
      </IconButton>
    </div>
  </div>
  <div
    className="rhs-filter-popover"
  >
    <div
      className="d-flex align-item-center margin-bottom-15 toggle-class"
    >
      <ToggleSwitch
        active={false}
        label="Show all subscriptions"
        labelPositioning="right"
        onChange={[Function]}
      />
    </div>
    <div
      className="margin-bottom-15"
    >
      <Dropdown
        disabled={false}
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Me",
              "value": "me",
            },
            Object {
              "label": "Anyone",
              "value": "anyone",
            },
          ]
        }
        placeholder="Created By"
        value="mockUser"
      />
    </div>
    <div
      className="text-align-right"
    >
      <Button
        extraClass="margin-right-8"
        isDisabled={false}
        isSecondaryButton={true}
        onClick={[Function]}
        text="Reset"
      />
      <Button
        onClick={[Function]}
        text="Close"
      />
    </div>
  </div>
  <div
    className="rhs-filter-popover rhs-menu-popover"
  >
    <MenuButtons
      buttons={
        Array [
          Object {
            "icon": <SVGWrapper
              height={16}
              viewBox="0 0 17 12"
              width={16}
            >
              <path
                d="M14.4 0H1.6C0.72 0 0 0.72 0 1.6V12.8C0 13.68 0.72 14.4 1.6 14.4H14.4C15.28 14.4 16 13.68 16 12.8V1.6C16 0.72 15.28 0 14.4 0ZM14.4 12.8H1.6V3.2H14.4V12.8ZM6.4 7.2V5.6H12.8V7.2H6.4ZM3.2 5.6H4.8V7.2H3.2V5.6ZM6.4 10.4V8.8H12.8V10.4H6.4ZM3.2 8.8H4.8V10.4H3.2V8.8Z"
                fill="#3F4350"
              />
            </SVGWrapper>,
            "onClick": [Function],
            "text": "Begin catalog request",
          },
          Object {
            "icon": <SVGWrapper
              height={16}
              viewBox="0 0 17 17"
              width={16}
            >
              <path
                d="M9.00002 0.666626C4.41669 0.666626 0.666687 4.41663 0.666687 8.99996C0.666687 13.5833 4.41669 17.3333 9.00002 17.3333C13.5834 17.3333 17.3334 13.5833 17.3334 8.99996C17.3334 4.41663 13.5834 0.666626 9.00002 0.666626ZM9.00002 15.6666C5.33335 15.6666 2.33335 12.6666 2.33335 8.99996C2.33335 5.33329 5.33335 2.33329 9.00002 2.33329C12.6667 2.33329 15.6667 5.33329 15.6667 8.99996C15.6667 12.6666 12.6667 15.6666 9.00002 15.6666ZM9.41752 9.83329H8.58419L8.16752 4.83329H9.83419L9.41752 9.83329ZM9.83335 12.3333C9.83335 12.7933 9.46002 13.1666 9.00002 13.1666C8.54002 13.1666 8.16669 12.7933 8.16669 12.3333C8.16669 11.8733 8.54002 11.5 9.00002 11.5C9.46002 11.5 9.83335 11.8733 9.83335 12.3333Z"
                fill="#3F4350"
              />
            </SVGWrapper>,
            "onClick": [Function],
            "text": "Create an incident",
          },
          Object {
            "icon": <SVGWrapper
              height={14}
              viewBox="0 2 16 10"
              width={14}
            >
              <path
                d="M11.6667 10.9511C11.0756 10.9511 10.5467 11.1844 10.1422 11.55L4.59667 8.32222C4.63556 8.14333 4.66667 7.96444 4.66667 7.77778C4.66667 7.59111 4.63556 7.41222 4.59667 7.23333L10.08 4.03667C10.5 4.42556 11.0522 4.66667 11.6667 4.66667C12.9578 4.66667 14 3.62444 14 2.33333C14 1.04222 12.9578 0 11.6667 0C10.3756 0 9.33333 1.04222 9.33333 2.33333C9.33333 2.52 9.36444 2.69889 9.40333 2.87778L3.92 6.07444C3.5 5.68556 2.94778 5.44444 2.33333 5.44444C1.04222 5.44444 0 6.48667 0 7.77778C0 9.06889 1.04222 10.1111 2.33333 10.1111C2.94778 10.1111 3.5 9.87 3.92 9.48111L9.45778 12.7089C9.41889 12.8722 9.39556 13.0433 9.39556 13.2222C9.39556 14.4744 10.4144 15.4856 11.6667 15.4856C12.9189 15.4856 13.9378 14.4744 13.9378 13.2222C13.9378 11.97 12.9189 10.9511 11.6667 10.9511ZM11.6667 1.55556C12.0944 1.55556 12.4444 1.90556 12.4444 2.33333C12.4444 2.76111 12.0944 3.11111 11.6667 3.11111C11.2389 3.11111 10.8889 2.76111 10.8889 2.33333C10.8889 1.90556 11.2389 1.55556 11.6667 1.55556ZM2.33333 8.55556C1.90556 8.55556 1.55556 8.20555 1.55556 7.77778C1.55556 7.35 1.90556 7 2.33333 7C2.76111 7 3.11111 7.35 3.11111 7.77778C3.11111 8.20555 2.76111 8.55556 2.33333 8.55556ZM11.6667 14C11.2389 14 10.8889 13.65 10.8889 13.2222C10.8889 12.7944 11.2389 12.4444 11.6667 12.4444C12.0944 12.4444 12.4444 12.7944 12.4444 13.2222C12.4444 13.65 12.0944 14 11.6667 14Z"
                fill="#3F4350"
              />
            </SVGWrapper>,
            "onClick": [Function],
            "text": "Share a record",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
